### ISSUE TITLE
Byobu shortcuts stop working

### ISSUE THREAD

**Issue by Khalian** [jwilm/alacritty#360]
Terminal emulator : Alacritty
Terminal multiplexer : byobu
System : Mac OS X El Capitan

Hi

I just downloaded and installed alacritty. The byobu shortcuts for creating and deleting shortcuts was fine for a good hour before it stopped working unexpectedly. There is absolutely no information on stdout or anywhere in logs. Please help. 

P.S : Said shortcuts work fine on both iterm2 and terminal.app. I was under the impression that alacritty's SLA was "just works multiplexer supporter".

**Comment by jwilm** [comment]
Did Alacritty happen to jump to 100% CPU at this time? There's a bug in the Cocoa part of the windowing library we depend on that causes the event loop to get stuck.

**Comment by Khalian** [comment]
I think so it did. I was at pretty high a load at the time when it surfaced. However its stuck there ever since. Do I need to restart the process ? (I presume I do, since you state its an infinite event loop).

**Comment by Khalian** [comment]
Ok I tried restarting alacritty, did not work.

**Comment by jwilm** [comment]
Alacritty has a flag `--print-events` that you can run with. This should will show a bunch of window events, but importantly, it will show the keypress events. Would you mind trying to run with that and see if your key press events are actually registering?

**Comment by pExeen** [comment]
same problem on arch linux. here is the output of --print-events 👍 
`glutin event: KeyboardInput(Pressed, 37, Some(LControl), , None)
glutin event: KeyboardInput(Pressed, 68, Some(F2), CONTROL, None)
glutin event: KeyboardInput(Released, 68, Some(F2), CONTROL, None)
glutin event: KeyboardInput(Released, 37, Some(LControl), CONTROL, None)
glutin event: Awakened
glutin event: Awakened
glutin event: KeyboardInput(Pressed, 50, Some(LShift), , None)
glutin event: KeyboardInput(Pressed, 68, Some(F2), SHIFT, None)
glutin event: KeyboardInput(Released, 68, Some(F2), SHIFT, None)
glutin event: KeyboardInput(Released, 50, Some(LShift), SHIFT, None)
`

**Comment by jwilm** [comment]
@pExeen did the events stop showing up when the byobu shortcuts stopped working?

**Comment by pExeen** [comment]
no they are sitell showing up

**Comment by pExeen** [comment]
i forget to say that f2 is working to do a new window

**Comment by Khalian** [comment]
Nothing works for me (other than terminal swtiches). I am going to stop experimenting on this for now.

**Comment by ChrisMacNaughton** [comment]
I'm having this same issue on Ubuntu. f2 _does_ work for creating a new byobu window, but SHIFT-F2 and CTRL-F2 do not work for splitting the window

**Comment by thypon** [comment]
Same for me on Linux.

**Comment by vkorchagin** [comment]
I found that split resizing doesn't work in byobu under alacritty. It seems that Alacritty sends wrong escape sequences for "Shit+Alt+Up|Down|Right|Left". The workaround is to add the following key bindings to the config:
```
  # Byobu split resize sequences.
  - { key: Up,       mods: Shift|Alt,   chars: "\x1b[1;4A"                   }
  - { key: Down,     mods: Shift|Alt,   chars: "\x1b[1;4B"                   }
  - { key: Right,    mods: Shift|Alt,   chars: "\x1b[1;4C"                   }
  - { key: Left,     mods: Shift|Alt,   chars: "\x1b[1;4D"                   }
```
I took the sequences from https://stackoverflow.com/questions/26180096/os-x-byobu-vertical-split

**Comment by chrisduerr** [comment]
@vkorchagin this issue should be addressed with the #2688 issue. Can you confirm if this issue itself is still present? It doesn't sound familiar so I'd expect this issue to be resolved.

**Comment by vkorchagin** [comment]
Thanks for the pointer! The problem I mentioned looks like a different issue indeed.

As for this one, I can't reproduce it. I use byobu with alacritty on a regular basis, and some terminal windows stay open for days. I've never seen the behavior described above.
